// tinymce-a11y-checker/lib/th/modules/rules/img-alt-length.js
import formatMessage from "../format-message";
const MAX_ALT_LENGTH = 120;
export default {
  "max-alt-length": MAX_ALT_LENGTH,
  id: "img-alt-length",
  test: elem => {
    if (elem.tagName !== "IMG") {
      return true;
    }

    const alt = elem.getAttribute("alt");
    return alt == null || alt.length <= MAX_ALT_LENGTH;
  },
  data: elem => {
    const alt = elem.getAttribute("alt");
    return {
      alt: alt || ""
    };
  },
  form: () => [{
    label: "\u0E40\u0E1B\u0E25\u0E35\u0E48\u0E22\u0E19\u0E02\u0E49\u0E2D\u0E04\u0E27\u0E32\u0E21\u0E40\u0E1C\u0E37\u0E48\u0E2D\u0E40\u0E25\u0E37\u0E2D\u0E01",
    dataKey: "alt",
    textarea: true
  }],
  update: (elem, data) => {
    elem.setAttribute("alt", data.alt);
    return elem;
  },
  message: () => "\u0E02\u0E49\u0E2D\u0E04\u0E27\u0E32\u0E21\u0E04\u0E38\u0E13\u0E25\u0E31\u0E01\u0E29\u0E13\u0E30\u0E40\u0E1C\u0E37\u0E48\u0E2D\u0E40\u0E25\u0E37\u0E2D\u0E01\u0E44\u0E21\u0E48\u0E04\u0E27\u0E23\u0E22\u0E32\u0E27\u0E40\u0E01\u0E34\u0E19\u0E01\u0E27\u0E48\u0E32 120 \u0E15\u0E31\u0E27\u0E2D\u0E31\u0E01\u0E29\u0E23",
  why: () => "\u0E15\u0E31\u0E27\u0E2D\u0E48\u0E32\u0E19\u0E2B\u0E19\u0E49\u0E32\u0E08\u0E2D\u0E44\u0E21\u0E48\u0E2A\u0E32\u0E21\u0E32\u0E23\u0E16\u0E23\u0E30\u0E1A\u0E38\u0E40\u0E19\u0E37\u0E49\u0E2D\u0E2B\u0E32\u0E17\u0E35\u0E48\u0E08\u0E31\u0E14\u0E41\u0E2A\u0E14\u0E07\u0E43\u0E19\u0E20\u0E32\u0E1E\u0E2B\u0E32\u0E01\u0E44\u0E21\u0E48\u0E35\u0E02\u0E49\u0E2D\u0E04\u0E27\u0E32\u0E21\u0E40\u0E1C\u0E37\u0E48\u0E2D\u0E40\u0E25\u0E37\u0E2D\u0E01\u0E40\u0E1E\u0E37\u0E48\u0E2D\u0E23\u0E30\u0E1A\u0E38\u0E1A\u0E23\u0E34\u0E1A\u0E17\u0E41\u0E25\u0E30\u0E04\u0E27\u0E32\u0E21\u0E2B\u0E21\u0E32\u0E22\u0E02\u0E2D\u0E07\u0E20\u0E32\u0E1E \u0E02\u0E49\u0E2D\u0E04\u0E27\u0E32\u0E21\u0E40\u0E1C\u0E37\u0E48\u0E2D\u0E40\u0E25\u0E37\u0E2D\u0E01\u0E04\u0E27\u0E23\u0E01\u0E23\u0E30\u0E0A\u0E31\u0E1A\u0E41\u0E25\u0E30\u0E40\u0E02\u0E49\u0E32\u0E43\u0E08\u0E44\u0E14\u0E49\u0E07\u0E48\u0E32\u0E22",
  link: ""
};
