// tinymce-a11y-checker/lib/th/modules/rules/large-text-contrast.js
import formatMessage from "../format-message";
import contrast from "wcag-element-contrast";
import smallTextContrast from "./small-text-contrast";
import { onlyContainsLink, hasTextNode } from "../utils/dom";
export default {
  id: "large-text-contrast",
  test: (elem, config = {}) => {
    const disabled = config.disableContrastCheck == true;
    const noText = !hasTextNode(elem);

    if (disabled || noText || onlyContainsLink(elem) || !contrast.isLargeText(elem)) {
      return true;
    }

    return contrast(elem);
  },
  data: smallTextContrast.data,
  form: smallTextContrast.form,
  update: smallTextContrast.update,
  message: () => "\u0E02\u0E49\u0E2D\u0E04\u0E27\u0E32\u0E21\u0E17\u0E35\u0E48\u0E43\u0E2B\u0E0D\u0E48\u0E01\u0E27\u0E48\u0E32 18pt (\u0E2B\u0E23\u0E37\u0E2D\u0E15\u0E31\u0E27\u0E2B\u0E19\u0E32 14pt) \u0E04\u0E27\u0E23\u0E21\u0E35\u0E23\u0E30\u0E14\u0E31\u0E1A\u0E04\u0E27\u0E32\u0E21\u0E40\u0E1B\u0E23\u0E35\u0E22\u0E1A\u0E15\u0E48\u0E32\u0E07\u0E02\u0E31\u0E49\u0E19\u0E15\u0E48\u0E33\u0E17\u0E35\u0E48 3:1",
  why: () => "Text is difficult to read without sufficient contrast between the text and the background, especially for those with low vision.",
  link: "https://www.w3.org/TR/WCAG20-TECHS/G17.html",
  linkText: () => "Learn more about color contrast"
};
