// tinymce-a11y-checker/lib/th/modules/rules/table-header-scope.js
import formatMessage from "../format-message";
const VALID_SCOPES = ["row", "col", "rowgroup", "colgroup"];
export default {
  id: "table-header-scope",
  test: elem => {
    if (elem.tagName !== "TH") {
      return true;
    }

    return VALID_SCOPES.indexOf(elem.getAttribute("scope")) !== -1;
  },
  data: elem => {
    return {
      scope: elem.getAttribute("scope") || "none"
    };
  },
  form: () => [{
    label: "\u0E01\u0E33\u0E2B\u0E19\u0E14\u0E02\u0E2D\u0E1A\u0E40\u0E02\u0E15\u0E2B\u0E31\u0E27\u0E40\u0E23\u0E37\u0E48\u0E2D\u0E07",
    dataKey: "scope",
    options: [["none", "\u0E44\u0E21\u0E48\u0E21\u0E35"], ["row", "\u0E41\u0E16\u0E27"], ["col", "\u0E04\u0E2D\u0E25\u0E31\u0E21\u0E19\u0E4C"], ["rowgroup", "\u0E01\u0E25\u0E38\u0E48\u0E21\u0E41\u0E16\u0E27"], ["colgroup", "\u0E01\u0E25\u0E38\u0E48\u0E21\u0E04\u0E2D\u0E25\u0E31\u0E21\u0E19\u0E4C"]]
  }],
  update: (elem, data) => {
    if (data.header === "none") {
      elem.removeAttribute("scope");
    } else {
      elem.setAttribute("scope", data.scope);
    }

    return elem;
  },
  message: () => "\u0E2B\u0E31\u0E27\u0E40\u0E23\u0E37\u0E48\u0E2D\u0E07\u0E15\u0E32\u0E23\u0E32\u0E07\u0E04\u0E27\u0E23\u0E23\u0E30\u0E1A\u0E38\u0E02\u0E2D\u0E1A\u0E40\u0E02\u0E15",
  why: () => "\u0E15\u0E31\u0E27\u0E2D\u0E48\u0E32\u0E19\u0E2B\u0E19\u0E49\u0E32\u0E08\u0E2D\u0E44\u0E21\u0E48\u0E2A\u0E32\u0E21\u0E32\u0E23\u0E16\u0E41\u0E1B\u0E25\u0E04\u0E27\u0E32\u0E21\u0E2B\u0E21\u0E32\u0E22\u0E15\u0E32\u0E23\u0E32\u0E07\u0E42\u0E14\u0E22\u0E44\u0E21\u0E48\u0E21\u0E35\u0E42\u0E04\u0E23\u0E07\u0E2A\u0E23\u0E49\u0E32\u0E07\u0E17\u0E35\u0E48\u0E16\u0E39\u0E01\u0E15\u0E49\u0E2D\u0E07 \u0E2B\u0E31\u0E27\u0E40\u0E23\u0E37\u0E48\u0E2D\u0E07\u0E15\u0E32\u0E23\u0E32\u0E07\u0E23\u0E30\u0E1A\u0E38\u0E17\u0E34\u0E28\u0E17\u0E32\u0E07\u0E41\u0E25\u0E30\u0E02\u0E2D\u0E1A\u0E40\u0E02\u0E15\u0E02\u0E2D\u0E07\u0E40\u0E19\u0E37\u0E49\u0E2D\u0E2B\u0E32",
  link: "https://www.w3.org/TR/WCAG20-TECHS/H63.html",
  linkText: () => "Learn more about using scope attributes with tables"
};
