// @instructure/ui-avatar/lib/Avatar/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Avatar = exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _createSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/createSuper"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames2 = _interopRequireDefault(require("classnames"));

var _View = require("@instructure/ui-view/lib/View");

var _themeable = require("@instructure/ui-themeable/lib/themeable.js");

var _ThemeablePropTypes = require("@instructure/ui-themeable/lib/ThemeablePropTypes.js");

var _passthroughProps = require("@instructure/ui-react-utils/lib/passthroughProps.js");

var _deprecated = require("@instructure/ui-react-utils/lib/deprecated.js");

var _callRenderProp = require("@instructure/ui-react-utils/lib/callRenderProp.js");

var _testable = require("@instructure/ui-testable/lib/testable.js");

var _theme = _interopRequireDefault(require("./theme.js"));

var _dec, _dec2, _dec3, _class, _class2, _temp;

var styles = {
  componentId: 'elMgC',
  template: function template(theme) {
    return "\n\n.elMgC_bGBk{background-clip:content-box;background-color:".concat(theme.background || 'inherit', ";background-position:50%;background-repeat:no-repeat;background-size:cover;box-sizing:border-box;height:2.5em;line-height:0;overflow:hidden;text-align:center}\n\n[dir=ltr] .elMgC_bGBk,[dir=rtl] .elMgC_bGBk{text-align:center}\n\n.elMgC_bGBk.elMgC_uUeq{font-size:inherit}\n\n.elMgC_bGBk.elMgC_uUeq,.elMgC_bGBk.elMgC_UKsG{border-width:").concat(theme.borderWidthSmall || 'inherit', "}\n\n.elMgC_bGBk.elMgC_UKsG{font-size:0.5rem}\n\n.elMgC_bGBk.elMgC_dfBC{font-size:0.75rem}\n\n.elMgC_bGBk.elMgC_doqw,.elMgC_bGBk.elMgC_dfBC{border-width:").concat(theme.borderWidthSmall || 'inherit', "}\n\n.elMgC_bGBk.elMgC_doqw{font-size:1rem}\n\n.elMgC_bGBk.elMgC_ycrn{font-size:1.25rem}\n\n.elMgC_bGBk.elMgC_cMDj,.elMgC_bGBk.elMgC_ycrn{border-width:").concat(theme.borderWidthMedium || 'inherit', "}\n\n.elMgC_bGBk.elMgC_cMDj{font-size:1.5rem}\n\n.elMgC_bGBk.elMgC_eoMd{font-size:1.75rem}\n\n.elMgC_bGBk.elMgC_eoMd,.elMgC_bGBk.elMgC_fdca{border-width:").concat(theme.borderWidthMedium || 'inherit', "}\n\n.elMgC_bGBk.elMgC_fdca{font-size:2rem}\n\n.elMgC_bGBk.elMgC_cHTY{border:0;box-shadow:inset 0 0 ").concat(theme.boxShadowBlur || 'inherit', " 0 ").concat(theme.boxShadowColor || 'inherit', "}\n\n.elMgC_bGBk:not(.elMgC_cHTY){border-color:").concat(theme.borderColor || 'inherit', ";border-style:solid}\n\n.elMgC_bGBk:not(.elMgC_cHTY).elMgC_fKgA{background-clip:border-box;border:0;padding:0}\n\n.elMgC_bGBk:not(.elMgC_cHTY).elMgC_fKgA.elMgC_uUeq,.elMgC_bGBk:not(.elMgC_cHTY).elMgC_fKgA.elMgC_doqw,.elMgC_bGBk:not(.elMgC_cHTY).elMgC_fKgA.elMgC_dfBC,.elMgC_bGBk:not(.elMgC_cHTY).elMgC_fKgA.elMgC_UKsG{padding:").concat(theme.borderWidthSmall || 'inherit', "}\n\n.elMgC_bGBk:not(.elMgC_cHTY).elMgC_fKgA.elMgC_cMDj,.elMgC_bGBk:not(.elMgC_cHTY).elMgC_fKgA.elMgC_ycrn,.elMgC_bGBk:not(.elMgC_cHTY).elMgC_fKgA.elMgC_eoMd,.elMgC_bGBk:not(.elMgC_cHTY).elMgC_fKgA.elMgC_fdca{padding:").concat(theme.borderWidthMedium || 'inherit', "}\n\n.elMgC_eYKj{font-family:").concat(theme.fontFamily || 'inherit', ";font-weight:").concat(theme.fontWeight || 'inherit', ";letter-spacing:0.0313em;line-height:2.375em}\n\n.elMgC_ddES{color:").concat(theme.color || 'inherit', "}\n\n.elMgC_ekuM{color:").concat(theme.colorShamrock || 'inherit', "}\n\n.elMgC_bXnN{color:").concat(theme.colorBarney || 'inherit', "}\n\n.elMgC_byip{color:").concat(theme.colorCrimson || 'inherit', "}\n\n.elMgC_dKVo{color:").concat(theme.colorFire || 'inherit', "}\n\n.elMgC_eaOe{color:").concat(theme.colorLicorice || 'inherit', "}\n\n.elMgC_fuVp{color:").concat(theme.colorAsh || 'inherit', "}\n\n.elMgC_fKgA{color:").concat(theme.background || 'inherit', "}\n\n.elMgC_fKgA.elMgC_ddES{background-color:").concat(theme.color || 'inherit', "}\n\n.elMgC_fKgA.elMgC_ekuM{background-color:").concat(theme.colorShamrock || 'inherit', "}\n\n.elMgC_fKgA.elMgC_bXnN{background-color:").concat(theme.colorBarney || 'inherit', "}\n\n.elMgC_fKgA.elMgC_byip{background-color:").concat(theme.colorCrimson || 'inherit', "}\n\n.elMgC_fKgA.elMgC_dKVo{background-color:").concat(theme.colorFire || 'inherit', "}\n\n.elMgC_fKgA.elMgC_eaOe{background-color:").concat(theme.colorLicorice || 'inherit', "}\n\n.elMgC_fKgA.elMgC_fuVp{background-color:").concat(theme.colorAsh || 'inherit', "}\n\n.elMgC_MrJH{display:none}\n\n.elMgC_cJVF{border-radius:100%;overflow:hidden;position:relative;width:2.5em}\n\n.elMgC_bRWf{width:3em}\n\n.elMgC_eoCh{align-items:center;display:flex;height:100%;justify-content:center;width:100%}\n\n.elMgC_eoCh svg{fill:currentColor;height:1em;width:1em}");
  },
  'root': 'elMgC_bGBk',
  'auto': 'elMgC_uUeq',
  'xx-small': 'elMgC_UKsG',
  'x-small': 'elMgC_dfBC',
  'small': 'elMgC_doqw',
  'medium': 'elMgC_ycrn',
  'large': 'elMgC_cMDj',
  'x-large': 'elMgC_eoMd',
  'xx-large': 'elMgC_fdca',
  'loaded': 'elMgC_cHTY',
  'hasInverseColor': 'elMgC_fKgA',
  'initials': 'elMgC_eYKj',
  'default': 'elMgC_ddES',
  'shamrock': 'elMgC_ekuM',
  'barney': 'elMgC_bXnN',
  'crimson': 'elMgC_byip',
  'fire': 'elMgC_dKVo',
  'licorice': 'elMgC_eaOe',
  'ash': 'elMgC_fuVp',
  'loadImage': 'elMgC_MrJH',
  'circle': 'elMgC_cJVF',
  'rectangle': 'elMgC_bRWf',
  'iconSVG': 'elMgC_eoCh'
};

/**
---
category: components
---
**/
var Avatar = (_dec = (0, _testable.testable)(), _dec2 = (0, _deprecated.deprecated)('8.0.0', {
  inline: 'display',
  variant: 'shape'
}), _dec3 = (0, _themeable.themeable)(_theme.default, styles), _dec(_class = _dec2(_class = _dec3(_class = (_temp = _class2 = /*#__PURE__*/function (_Component) {
  (0, _inherits2.default)(Avatar, _Component);

  var _super = (0, _createSuper2.default)(Avatar);

  function Avatar() {
    var _this;

    (0, _classCallCheck2.default)(this, Avatar);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));
    _this.state = {
      loaded: false
    };

    _this.handleImageLoaded = function (event) {
      _this.setState({
        loaded: true
      });

      _this.props.onImageLoaded(event);
    };

    return _this;
  }

  (0, _createClass2.default)(Avatar, [{
    key: "componentDidUpdate",
    value: function componentDidUpdate() {
      // in case the image is unset in an update, show icons/initials again
      if (this.state.loaded && !this.props.src) {
        this.setState({
          loaded: false
        });
      }
    }
  }, {
    key: "makeInitialsFromName",
    value: function makeInitialsFromName() {
      var name = this.props.name;

      if (!name || typeof name !== 'string') {
        return;
      }

      name = name.trim();

      if (name.length === 0) {
        return;
      }

      if (name.match(/\s+/)) {
        var names = name.split(/\s+/);
        return (names[0][0] + names[names.length - 1][0]).toUpperCase();
      } else {
        return name[0].toUpperCase();
      }
    }
  }, {
    key: "renderLoadImage",
    value: function renderLoadImage() {
      // This image element is visually hidden and is here for loading purposes only
      return /*#__PURE__*/_react.default.createElement("img", {
        src: this.props.src,
        className: styles.loadImage,
        alt: this.props.alt,
        onLoad: this.handleImageLoaded,
        "aria-hidden": "true"
      });
    }
  }, {
    key: "renderInitials",
    value: function renderInitials() {
      return /*#__PURE__*/_react.default.createElement("span", {
        className: styles.initials,
        "aria-hidden": "true"
      }, this.makeInitialsFromName());
    }
  }, {
    key: "renderContent",
    value: function renderContent() {
      var renderIcon = this.props.renderIcon;

      if (!renderIcon) {
        return this.renderInitials();
      }

      return /*#__PURE__*/_react.default.createElement("span", {
        className: styles.iconSVG
      }, (0, _callRenderProp.callRenderProp)(renderIcon));
    }
  }, {
    key: "render",
    value: function render() {
      var _classnames;

      var _this$props = this.props,
          onImageLoaded = _this$props.onImageLoaded,
          props = (0, _objectWithoutProperties2.default)(_this$props, ["onImageLoaded"]);
      var loaded = this.state.loaded;
      return /*#__PURE__*/_react.default.createElement(_View.View, Object.assign({}, (0, _passthroughProps.passthroughProps)(props), {
        style: {
          backgroundImage: loaded ? "url('".concat(this.props.src, "')") : void 0
        },
        className: (0, _classnames2.default)((_classnames = {}, (0, _defineProperty2.default)(_classnames, styles.root, true), (0, _defineProperty2.default)(_classnames, styles[this.props.size], true), (0, _defineProperty2.default)(_classnames, styles[this.props.variant || this.props.shape], true), (0, _defineProperty2.default)(_classnames, styles[this.props.color], true), (0, _defineProperty2.default)(_classnames, styles.loaded, loaded), (0, _defineProperty2.default)(_classnames, styles.hasInverseColor, this.props.hasInverseColor), _classnames)),
        "aria-label": this.props.alt ? this.props.alt : null,
        role: this.props.alt ? 'img' : null,
        as: this.props.as,
        elementRef: this.props.elementRef,
        margin: this.props.margin,
        display: this.props.display === 'block' || this.props.inline === false ? 'block' : 'inline-block'
      }), this.renderLoadImage(), !loaded && this.renderContent());
    }
  }]);
  Avatar.displayName = "Avatar";
  return Avatar;
}(_react.Component), _class2.propTypes = {
  name: _propTypes.default.string.isRequired,

  /*
   * URL of the image to display as the background image
   */
  src: _propTypes.default.string,

  /*
   * Accessible label
   */
  alt: _propTypes.default.string,
  size: _propTypes.default.oneOf(['auto', 'xx-small', 'x-small', 'small', 'medium', 'large', 'x-large', 'xx-large']),
  color: _propTypes.default.oneOf(['default', 'shamrock', 'barney', 'crimson', 'fire', 'licorice', 'ash']),

  /**
   * In inverse color mode the background and text/icon colors are inverted
   */
  hasInverseColor: _propTypes.default.bool,
  shape: _propTypes.default.oneOf(['circle', 'rectangle']),

  /**
   * Valid values are `0`, `none`, `auto`, `xxx-small`, `xx-small`, `x-small`,
   * `small`, `medium`, `large`, `x-large`, `xx-large`. Apply these values via
   * familiar CSS-like shorthand. For example: `margin="small auto large"`.
   */
  margin: _ThemeablePropTypes.ThemeablePropTypes.spacing,
  display: _propTypes.default.oneOf(['inline-block', 'block']),

  /**
   * Callback fired when the avatar image has loaded
   */
  onImageLoaded: _propTypes.default.func,

  /**
   * the element type to render as
   */
  as: _propTypes.default.elementType,
  // eslint-disable-line react/require-default-props

  /**
   * provides a reference to the underlying html element
   */
  elementRef: _propTypes.default.func,

  /**
   * An icon, or function that returns an icon that gets displayed. If the `src` prop is provided, `src` will have priority.
   */
  renderIcon: _propTypes.default.oneOfType([_propTypes.default.node, _propTypes.default.func]),

  /* eslint-disable react/require-default-props */

  /**
   * __Deprecated - use `display`__
   */
  inline: _propTypes.default.bool,

  /**
   * __Deprecated - use `shape`__
   */
  variant: _propTypes.default.oneOf(['circle', 'rectangle'])
  /* eslint-enable react/require-default-props */

}, _class2.defaultProps = {
  src: void 0,
  alt: void 0,
  margin: void 0,
  elementRef: void 0,
  size: 'medium',
  color: 'default',
  hasInverseColor: false,
  shape: 'circle',
  display: 'inline-block',
  onImageLoaded: function onImageLoaded(event) {},
  renderIcon: null
}, _temp)) || _class) || _class) || _class);
exports.Avatar = Avatar;
var _default = Avatar;
exports.default = _default;
