// @instructure/ui-modal/es/Modal/ModalHeader/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _createSuper from "@babel/runtime/helpers/esm/createSuper";

var _dec, _dec2, _class, _class2, _temp;

/*
 * The MIT License (MIT)
 *
 * Copyright (c) 2015 - present Instructure, Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { matchComponentTypes, passthroughProps } from '@instructure/ui-react-utils';
import { CloseButton } from '@instructure/ui-buttons';
import { themeable } from '@instructure/ui-themeable';
import { testable } from '@instructure/ui-testable';
var styles = {
  componentId: 'eHzxc',
  template: function template(theme) {
    return "\n\n.eHzxc_bGBk{background:".concat(theme.background || 'inherit', ";border-bottom-color:").concat(theme.borderColor || 'inherit', ";border-bottom-style:solid;border-bottom-width:0.0625rem;box-sizing:border-box;flex:0 0 auto}\n\n.eHzxc_enfx{background:").concat(theme.inverseBackground || 'inherit', ";border-bottom-color:").concat(theme.inverseBorderColor || 'inherit', "}\n\n.eHzxc_bcOZ{-webkit-padding-end:").concat(theme.padding || 'inherit', ";-webkit-padding-start:").concat(theme.padding || 'inherit', ";padding:").concat(theme.padding || 'inherit', ";padding-inline-end:").concat(theme.padding || 'inherit', ";padding-inline-start:").concat(theme.padding || 'inherit', "}\n\n[dir=ltr] .eHzxc_bcOZ{padding-left:").concat(theme.padding || 'inherit', ";padding-right:").concat(theme.padding || 'inherit', "}\n\n[dir=rtl] .eHzxc_bcOZ{padding-left:").concat(theme.padding || 'inherit', ";padding-right:").concat(theme.padding || 'inherit', "}\n\n.eHzxc_bcOZ.eHzxc_ssMr{-webkit-padding-end:calc(").concat(theme.padding || 'inherit', "*2 + 1em);padding-inline-end:calc(").concat(theme.padding || 'inherit', "*2 + 1em)}\n\n[dir=ltr] .eHzxc_bcOZ.eHzxc_ssMr{padding-right:calc(").concat(theme.padding || 'inherit', "*2 + 1em)}\n\n[dir=rtl] .eHzxc_bcOZ.eHzxc_ssMr{padding-left:calc(").concat(theme.padding || 'inherit', "*2 + 1em)}\n\n.eHzxc_cieE{-webkit-padding-end:").concat(theme.paddingCompact || 'inherit', ";-webkit-padding-start:").concat(theme.paddingCompact || 'inherit', ";padding:").concat(theme.paddingCompact || 'inherit', ";padding-inline-end:").concat(theme.paddingCompact || 'inherit', ";padding-inline-start:").concat(theme.paddingCompact || 'inherit', "}\n\n[dir=ltr] .eHzxc_cieE{padding-left:").concat(theme.paddingCompact || 'inherit', ";padding-right:").concat(theme.paddingCompact || 'inherit', "}\n\n[dir=rtl] .eHzxc_cieE{padding-left:").concat(theme.paddingCompact || 'inherit', ";padding-right:").concat(theme.paddingCompact || 'inherit', "}\n\n.eHzxc_cieE.eHzxc_ssMr{-webkit-padding-end:calc(").concat(theme.paddingCompact || 'inherit', "*2 + 1em);padding-inline-end:calc(").concat(theme.paddingCompact || 'inherit', "*2 + 1em)}\n\n[dir=ltr] .eHzxc_cieE.eHzxc_ssMr{padding-right:calc(").concat(theme.paddingCompact || 'inherit', "*2 + 1em)}\n\n[dir=rtl] .eHzxc_cieE.eHzxc_ssMr{padding-left:calc(").concat(theme.paddingCompact || 'inherit', "*2 + 1em)}");
  },
  'root': 'eHzxc_bGBk',
  'inverse': 'eHzxc_enfx',
  'defaultSpacing': 'eHzxc_bcOZ',
  'withCloseButton': 'eHzxc_ssMr',
  'compactSpacing': 'eHzxc_cieE'
};
import theme from "./theme.js";
/**
---
parent: Modal
id: Modal.Header
---
**/

var ModalHeader = (_dec = testable(), _dec2 = themeable(theme, styles), _dec(_class = _dec2(_class = (_temp = _class2 = /*#__PURE__*/function (_Component) {
  _inherits(ModalHeader, _Component);

  var _super = _createSuper(ModalHeader);

  function ModalHeader() {
    _classCallCheck(this, ModalHeader);

    return _super.apply(this, arguments);
  }

  _createClass(ModalHeader, [{
    key: "render",
    value: function render() {
      var _classes;

      var _this$props = this.props,
          children = _this$props.children,
          variant = _this$props.variant,
          spacing = _this$props.spacing,
          rest = _objectWithoutProperties(_this$props, ["children", "variant", "spacing"]);

      var usesCloseButton = false;
      React.Children.forEach(children, function (child) {
        if (child && matchComponentTypes(child, [CloseButton])) {
          usesCloseButton = true;
        }
      });
      var classes = (_classes = {}, _defineProperty(_classes, styles.root, true), _defineProperty(_classes, styles["".concat(spacing, "Spacing")], true), _defineProperty(_classes, styles.inverse, variant === 'inverse'), _defineProperty(_classes, styles.withCloseButton, usesCloseButton === true), _classes);
      return /*#__PURE__*/React.createElement("div", Object.assign({
        className: classnames(classes)
      }, passthroughProps(rest)), children);
    }
  }]);

  ModalHeader.displayName = "ModalHeader";
  return ModalHeader;
}(Component), _class2.propTypes = {
  children: PropTypes.node,
  variant: PropTypes.oneOf(['default', 'inverse']),
  spacing: PropTypes.oneOf(['default', 'compact'])
}, _class2.defaultProps = {
  children: null,
  variant: 'default',
  spacing: 'default'
}, _temp)) || _class) || _class);
export default ModalHeader;
export { ModalHeader };
