// tinymce-a11y-checker/lib/vi/modules/rules/img-alt.js
import formatMessage from "../format-message";
export default {
  id: "img-alt",
  test: elem => {
    if (elem.tagName !== "IMG") {
      return true;
    }

    const alt = elem.hasAttribute("alt") ? elem.getAttribute("alt") : null;
    return alt !== null;
  },
  data: elem => {
    const alt = elem.hasAttribute("alt") ? elem.getAttribute("alt") : null;
    const decorative = alt !== null && alt.replace(/\s/g, "") === "";
    return {
      alt: alt || "",
      decorative: decorative
    };
  },
  form: () => [{
    label: "Th\xEAm v\u0103n b\u1EA3n thay th\u1EBF cho h\xECnh \u1EA3nh",
    dataKey: "alt",
    disabledIf: data => data.decorative
  }, {
    label: "H\xECnh \u1EA3nh trang tr\xED",
    dataKey: "decorative",
    checkbox: true
  }],
  update: (elem, data) => {
    if (data.decorative) {
      elem.setAttribute("alt", "");
      elem.setAttribute("role", "presentation");
    } else {
      elem.setAttribute("alt", data.alt);
      elem.removeAttribute("role");
    }

    return elem;
  },
  message: () => "H\xECnh \u1EA3nh ph\u1EA3i ch\u1EE9a m\u1ED9t thu\u1ED9c t\xEDnh thay th\u1EBF m\xF4 t\u1EA3 n\u1ED9i dung h\xECnh \u1EA3nh \u0111\xF3.",
  why: () => "Tr\xECnh \u0111\u1ECDc m\xE0n h\xECnh kh\xF4ng th\u1EC3 x\xE1c \u0111\u1ECBnh n\u1ED9i d\u1EE5ng \u0111\u01B0\u1EE3c hi\u1EC3n th\u1ECB trong h\xECnh \u1EA3nh m\xE0 kh\xF4ng c\xF3 v\u0103n b\u1EA3n thay th\u1EBF, v\u1ED1n c\xF3 ch\u1EE9c n\u0103ng m\xF4 t\u1EA3 n\u1ED9i dung v\xE0 \xFD ngh\u0129a c\u1EE7a h\xECnh \u1EA3nh.",
  link: "https://www.w3.org/TR/WCAG20-TECHS/H37.html",
  linkText: () => "T\xECm hi\u1EC3u th\xEAm v\u1EC1 s\u1EED d\u1EE5ng v\u0103n b\u1EA3n thay th\u1EBF cho h\xECnh \u1EA3nh"
};
