// tinymce-a11y-checker/lib/vi/modules/rules/img-alt-length.js
import formatMessage from "../format-message";
const MAX_ALT_LENGTH = 120;
export default {
  "max-alt-length": MAX_ALT_LENGTH,
  id: "img-alt-length",
  test: elem => {
    if (elem.tagName !== "IMG") {
      return true;
    }

    const alt = elem.getAttribute("alt");
    return alt == null || alt.length <= MAX_ALT_LENGTH;
  },
  data: elem => {
    const alt = elem.getAttribute("alt");
    return {
      alt: alt || ""
    };
  },
  form: () => [{
    label: "Thay \u0111\u1ED5i v\u0103n b\u1EA3n thay th\u1EBF",
    dataKey: "alt",
    textarea: true
  }],
  update: (elem, data) => {
    elem.setAttribute("alt", data.alt);
    return elem;
  },
  message: () => "V\u0103n b\u1EA3n thu\u1ED9c t\xEDnh thay th\u1EBF kh\xF4ng \u0111\u01B0\u1EE3c ch\u1EE9a qu\xE1 120 k\xFD t\u1EF1.",
  why: () => "Tr\xECnh \u0111\u1ECDc m\xE0n h\xECnh kh\xF4ng th\u1EC3 x\xE1c \u0111\u1ECBnh n\u1ED9i d\u1EE5ng \u0111\u01B0\u1EE3c hi\u1EC3n th\u1ECB trong h\xECnh \u1EA3nh m\xE0 kh\xF4ng c\xF3 v\u0103n b\u1EA3n thay th\u1EBF, v\u1ED1n c\xF3 ch\u1EE9c n\u0103ng m\xF4 t\u1EA3 n\u1ED9i dung v\xE0 \xFD ngh\u0129a c\u1EE7a h\xECnh \u1EA3nh. V\u0103n b\u1EA3n thay th\u1EBF ph\u1EA3i \u0111\u01A1n gi\u1EA3n v\xE0 s\xFAc t\xEDch.",
  link: ""
};
