// tinymce-a11y-checker/lib/vi/modules/rules/paragraphs-for-headings.js
import formatMessage from "../format-message";
import { changeTag } from "../utils/dom";
const MAX_HEADING_LENGTH = 120;
const IS_HEADING = {
  H1: true,
  H2: true,
  H3: true,
  H4: true,
  H5: true,
  H6: true
};
export default {
  "max-heading-length": MAX_HEADING_LENGTH,
  id: "paragraphs-for-headings",
  test: elem => {
    if (!IS_HEADING[elem.tagName]) {
      return true;
    }

    return elem.textContent.length <= MAX_HEADING_LENGTH;
  },
  data: elem => {
    return {
      change: false
    };
  },
  form: () => [{
    label: "Thay \u0111\u1ED5i th\u1EBB \u0111\u1EA7u \u0111\u1EC1 cho \u0111o\u1EA1n",
    checkbox: true,
    dataKey: "change"
  }],
  update: (elem, data) => {
    let ret = elem;

    if (data.change) {
      ret = changeTag(elem, "p");
    }

    return ret;
  },
  message: () => "\u0110\u1EA7u \u0111\u1EC1 kh\xF4ng \u0111\u01B0\u1EE3c ch\u1EE9a qu\xE1 120 k\xFD t\u1EF1.",
  why: () => "Ng\u01B0\u1EDDi d\xF9ng kh\xF4ng b\u1ECB khi\u1EBFm th\u1ECB duy\u1EC7t nhanh trang web, t\xECm c\xE1c \u0111\u1EA7u \u0111\u1EC1 c\u1EE1 l\u1EDBn ho\u1EB7c \u0111\u01B0\u1EE3c in \u0111\u1EADm. Ng\u01B0\u1EDDi d\xF9ng tr\xECnh \u0111\u1ECDc m\xE0n h\xECnh ph\u1EE5 thu\u1ED9c v\xE0o \u0111\u1EA7u \u0111\u1EC1 \u0111\u1EC3 n\u1EAFm b\u1EAFt ng\u1EEF c\u1EA3nh. \u0110\u1EA7u \u0111\u1EC1 ph\u1EA3i s\xFAc t\xEDch trong ph\u1EA1m vi c\u1EA5u tr\xFAc ho\xE0n ch\u1EC9nh.",
  link: ""
};
