// tinymce-a11y-checker/lib/vi/rules/headings-sequence.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _formatMessage = _interopRequireDefault(require("../format-message"));

var _dom = require("../utils/dom");

/* Headings Sequence rule
 * this rule is ensuring that heading tags (H1-H6) are layed out in sequential
 * order for organizing your site.
 *
 * this rule only looks at H2-H6 headings. all other tags pass.
 * this rule will walk 'up-down' the dom to find the heading tag that is
 * laid out previous to the heading tag being checked.
 * this rule will see if the heading tag number of the previous heading is
 * one more than one less than it's own heading tag and will fail if so
 * this rule will check to see if there is no previous heading tag and will
 * fail the test if so
 */
const isHtag = elem => {
  const allHTags = {
    H1: true,
    H2: true,
    H3: true,
    H4: true,
    H5: true,
    H6: true
  };
  return elem && allHTags[elem.tagName] === true;
}; // gets the H tag that is furthest down in the tree from elem(inclusive)


const getHighestOrderHForElem = elem => {
  const allHForElem = Array.prototype.slice.call(elem.querySelectorAll("H1,H2,H3,H4,H5,H6"));

  if (allHForElem.length > 0) {
    return allHForElem.reverse()[0];
  }

  if (isHtag(elem)) {
    return elem;
  }

  return void 0;
}; // gets all siblings of elem that come before the elem ordered by nearest to
// elem


const getPrevSiblings = elem => {
  let ret = [];

  if (!elem || !elem.parentElement || !elem.parentElement.children) {
    return ret;
  }

  const sibs = elem.parentElement.children;

  for (var i = 0; i < sibs.length; i++) {
    if (sibs[i] === elem) {
      break;
    }

    ret.unshift(sibs[i]);
  }

  return ret;
};

const searchPrevSiblings = elem => {
  const sibs = getPrevSiblings(elem);
  let ret;

  for (let i = 0; i < sibs.length; i++) {
    ret = getHighestOrderHForElem(sibs[i]);

    if (ret) {
      break;
    }
  }

  return ret;
};

const _walkUpTree = elem => {
  let ret;

  if (!elem || elem.tagName === "BODY") {
    return void 0;
  }

  if (isHtag(elem)) {
    return elem;
  }

  ret = searchPrevSiblings(elem);

  if (!ret) {
    ret = _walkUpTree(elem.parentElement);
  }

  return ret;
};

const walkUpTree = elem => {
  let ret = searchPrevSiblings(elem);

  if (!ret) {
    ret = _walkUpTree(elem.parentElement);
  }

  return ret;
};

const getPriorHeading = elem => {
  return walkUpTree(elem);
}; // a valid prior H tag is greater or equal to one less than current


const getValidHeadings = elem => {
  const hNum = +elem.tagName.substring(1);
  const ret = {};

  for (var i = hNum - 1; i <= 6; i++) {
    ret[`H${i}`] = true;
  }

  return ret;
};

var _default = {
  id: "headings-sequence",
  test: elem => {
    const testTags = {
      H2: true,
      H3: true,
      H4: true,
      H5: true,
      H6: true
    };

    if (testTags[elem.tagName] !== true) {
      return true;
    }

    const validHeadings = getValidHeadings(elem);
    const priorHeading = getPriorHeading(elem);

    if (priorHeading) {
      return validHeadings[priorHeading.tagName];
    }

    return true;
  },
  data: elem => {
    return {
      action: "nothing"
    };
  },
  form: () => [{
    label: "Thao t\xE1c s\u1EBD th\u1EF1c hi\u1EC7n:",
    dataKey: "action",
    options: [["nothing", "\u0110\u1EC3 nguy\xEAn"], ["elem", "S\u1EEDa h\u1EC7 ph\xE2n c\u1EA5p \u0111\u1EA7u \u0111\u1EC1"], ["modify", "G\u1EE1 ki\u1EC3u \u0111\u1EA7u \u0111\u1EC1"]]
  }],
  update: (elem, data) => {
    if (!data || !data.action || data.action === "nothing") {
      return elem;
    }

    switch (data.action) {
      case "elem":
        {
          const priorH = getPriorHeading(elem);
          const hIdx = priorH ? +priorH.tagName.substring(1) : 0;
          return (0, _dom.changeTag)(elem, `H${hIdx + 1}`);
        }

      case "modify":
        {
          return (0, _dom.changeTag)(elem, "p");
        }
    }
  },
  message: () => "Kh\xF4ng \u0111\u01B0\u1EE3c b\u1ECF qua c\u1EA5p \u0111\u1EA7u \u0111\u1EC1.",
  why: () => "Ng\u01B0\u1EDDi d\xF9ng kh\xF4ng b\u1ECB khi\u1EBFm th\u1ECB duy\u1EC7t nhanh trang web, t\xECm c\xE1c \u0111\u1EA7u \u0111\u1EC1 c\u1EE1 l\u1EDBn ho\u1EB7c \u0111\u01B0\u1EE3c in \u0111\u1EADm. Ng\u01B0\u1EDDi d\xF9ng tr\xECnh \u0111\u1ECDc m\xE0n h\xECnh ph\u1EE5 thu\u1ED9c v\xE0o \u0111\u1EA7u \u0111\u1EC1 \u0111\u1EC3 n\u1EAFm b\u1EAFt ng\u1EEF c\u1EA3nh. \u0110\u1EA7u \u0111\u1EC1 ph\u1EA3i s\u1EED d\u1EE5ng c\u1EA5u tr\xFAc ho\xE0n ch\u1EC9nh.",
  link: "https://www.w3.org/TR/WCAG20-TECHS/G141.html",
  linkText: () => "T\xECm hi\u1EC3u th\xEAm v\u1EC1 s\u1EAFp x\u1EBFp c\xE1c ti\xEAu \u0111\u1EC1 trang"
};
exports.default = _default;
