// tinymce-a11y-checker/lib/vi/rules/img-alt-filename.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _formatMessage = _interopRequireDefault(require("../format-message"));

const FILENAMELIKE = /^(\S|\s)+\.\S+$/;
var _default = {
  id: "img-alt-filename",
  test: elem => {
    if (elem.tagName !== "IMG") {
      return true;
    }

    const alt = elem.hasAttribute("alt") ? elem.getAttribute("alt") : null;
    const isDecorative = alt !== null && alt.replace(/\s/g, "") === "";
    return !FILENAMELIKE.test(alt) || isDecorative;
  },
  data: elem => {
    const alt = elem.hasAttribute("alt") ? elem.getAttribute("alt") : null;
    const decorative = alt !== null && alt.replace(/\s/g, "") === "";
    return {
      alt: alt || "",
      decorative: decorative
    };
  },
  form: () => [{
    label: "Thay \u0111\u1ED5i v\u0103n b\u1EA3n thay th\u1EBF",
    dataKey: "alt",
    disabledIf: data => data.decorative
  }, {
    label: "H\xECnh \u1EA3nh trang tr\xED",
    dataKey: "decorative",
    checkbox: true
  }],
  update: (elem, data) => {
    if (data.decorative) {
      elem.setAttribute("alt", "");
      elem.setAttribute("role", "presentation");
    } else {
      elem.setAttribute("alt", data.alt);
      elem.removeAttribute("role");
    }

    return elem;
  },
  message: () => "Kh\xF4ng \u0111\u01B0\u1EE3c s\u1EED d\u1EE5ng t\xEAn t\u1EADp tin c\u1EE7a h\xECnh \u1EA3nh l\xE0m thu\u1ED9c t\xEDnh thay th\u1EBF m\xF4 t\u1EA3 n\u1ED9i dung h\xECnh \u1EA3nh.",
  why: () => "Tr\xECnh \u0111\u1ECDc m\xE0n h\xECnh kh\xF4ng th\u1EC3 x\xE1c \u0111\u1ECBnh n\u1ED9i d\u1EE5ng \u0111\u01B0\u1EE3c hi\u1EC3n th\u1ECB trong h\xECnh \u1EA3nh m\xE0 kh\xF4ng c\xF3 v\u0103n b\u1EA3n thay th\u1EBF v\xE0 t\xEAn t\u1EADp tin th\u01B0\u1EDDng l\xE0 chu\u1ED7i s\u1ED1 v\xE0 ch\u1EEF c\xE1i v\xF4 ngh\u0129a kh\xF4ng m\xF4 t\u1EA3 ng\u1EEF c\u1EA3nh ho\u1EB7c \xFD ngh\u0129a.",
  link: "https://www.w3.org/TR/WCAG20-TECHS/F30.html",
  linkText: () => "T\xECm hi\u1EC3u th\xEAm v\u1EC1 s\u1EED d\u1EE5ng t\xEAn t\u1EADp tin l\xE0m v\u0103n b\u1EA3n thay th\u1EBF"
};
exports.default = _default;
