// tinymce-a11y-checker/lib/vi/rules/small-text-contrast.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _formatMessage = _interopRequireDefault(require("../format-message"));

var _wcagElementContrast = _interopRequireDefault(require("wcag-element-contrast"));

var _dom = require("../utils/dom");

var _rgbHex = _interopRequireDefault(require("../utils/rgb-hex"));

var _default = {
  id: "small-text-contrast",
  test: (elem, config = {}) => {
    const disabled = config.disableContrastCheck == true;
    const noText = !(0, _dom.hasTextNode)(elem);

    if (disabled || noText || (0, _dom.onlyContainsLink)(elem) || _wcagElementContrast.default.isLargeText(elem)) {
      return true;
    }

    return (0, _wcagElementContrast.default)(elem);
  },
  data: elem => {
    const styles = window.getComputedStyle(elem);
    return {
      color: styles.color
    };
  },
  form: () => [{
    label: "Thay \u0111\u1ED5i m\xE0u v\u0103n b\u1EA3n",
    dataKey: "color",
    color: true
  }],
  update: (elem, data) => {
    elem.style.color = data.color;
    const styles = (0, _dom.splitStyleAttribute)(elem.getAttribute("data-mce-style") || "");

    if (data && data.color && data.color.indexOf("#") < 0) {
      styles.color = `#${(0, _rgbHex.default)(data.color)}`;
    } else {
      styles.color = data.color;
    }

    elem.setAttribute("data-mce-style", (0, _dom.createStyleString)(styles));
    return elem;
  },
  message: () => "V\u0103n b\u1EA3n c\xF3 c\u1EE1 nh\u1ECF h\u01A1n 18 pt (ho\u1EB7c in \u0111\u1EADm 14 pt) ph\u1EA3i hi\u1EC3n th\u1ECB t\u1EF7 l\u1EC7 t\u01B0\u01A1ng ph\u1EA3n t\u1ED1i thi\u1EC3u l\xE0 4,5:1.",
  why: () => "V\u0103n b\u1EA3n kh\xF3 \u0111\u1ECDc, kh\xF4ng \u0111\u1EE7 \u0111\u1ED9 t\u01B0\u01A1ng ph\u1EA3n gi\u1EEFa v\u0103n b\u1EA3n v\xE0 n\u1EC1n, \u0111\u1EB7c bi\u1EC7t l\xE0 \u0111\u1ED1i v\u1EDBi nh\u1EEFng ng\u01B0\u1EDDi c\xF3 th\u1ECB l\u1EF1c k\xE9m.",
  link: "https://www.w3.org/TR/WCAG20-TECHS/G17.html",
  linkText: () => "T\xECm hi\u1EC3u th\xEAm v\u1EC1 \u0111\u1ED9 t\u01B0\u01A1ng ph\u1EA3n m\xE0u s\u1EAFc"
};
exports.default = _default;
