// tinymce-a11y-checker/lib/vi/rules/table-header-scope.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _formatMessage = _interopRequireDefault(require("../format-message"));

const VALID_SCOPES = ["row", "col", "rowgroup", "colgroup"];
var _default = {
  id: "table-header-scope",
  test: elem => {
    if (elem.tagName !== "TH") {
      return true;
    }

    return VALID_SCOPES.indexOf(elem.getAttribute("scope")) !== -1;
  },
  data: elem => {
    return {
      scope: elem.getAttribute("scope") || "none"
    };
  },
  form: () => [{
    label: "C\xE0i \u0111\u1EB7t ph\u1EA1m vi \u0111\u1EA7u \u0111\u1EC1",
    dataKey: "scope",
    options: [["none", "Kh\xF4ng C\xF3"], ["row", "H\xE0ng"], ["col", "C\u1ED9t"], ["rowgroup", "Nh\xF3m h\xE0ng"], ["colgroup", "Nh\xF3m c\u1ED9t"]]
  }],
  update: (elem, data) => {
    if (data.header === "none") {
      elem.removeAttribute("scope");
    } else {
      elem.setAttribute("scope", data.scope);
    }

    return elem;
  },
  message: () => "\u0110\u1EA7u \u0111\u1EC1 b\u1EA3ng ph\u1EA3i x\xE1c \u0111\u1ECBnh ph\u1EA1m vi.",
  why: () => "Tr\xECnh \u0111\u1ECDc m\xE0n h\xECnh kh\xF4ng th\u1EC3 di\u1EC5n gi\u1EA3i b\u1EA3ng kh\xF4ng c\xF3 c\u1EA5u tr\xFAc ho\xE0n ch\u1EC9nh. \u0110\u1EA7u \u0111\u1EC1 b\u1EA3ng l\xE0 th\u1EE9 cung c\u1EA5p ph\u01B0\u01A1ng h\u01B0\u1EDBng v\xE0 ph\u1EA1m vi n\u1ED9i dung.",
  link: "https://www.w3.org/TR/WCAG20-TECHS/H63.html",
  linkText: () => "T\xECm hi\u1EC3u th\xEAm v\u1EC1 s\u1EED d\u1EE5ng thu\u1ED9c t\xEDnh ph\u1EA1m vi v\u1EDBi b\u1EA3ng"
};
exports.default = _default;
