// tinymce-a11y-checker/lib/vi/rules/table-caption.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _formatMessage = _interopRequireDefault(require("../format-message"));

var _dom = require("../utils/dom");

var _default = {
  id: "table-caption",
  test: elem => {
    if (elem.tagName !== "TABLE") {
      return true;
    }

    const caption = elem.querySelector("caption");
    return !!caption && caption.textContent.replace(/\s/g, "") !== "";
  },
  data: elem => {
    return {
      caption: ""
    };
  },
  form: () => [{
    label: "Th\xEAm ph\u1EE5 \u0111\u1EC1",
    dataKey: "caption"
  }],
  update: (elem, data) => {
    let caption = elem.querySelector("caption");

    if (!caption) {
      caption = elem.ownerDocument.createElement("caption");
      (0, _dom.prepend)(elem, caption);
    }

    caption.textContent = data.caption;
    return elem;
  },
  message: () => "B\u1EA3ng ph\u1EA3i c\xF3 ph\u1EE5 \u0111\u1EC1 m\xF4 t\u1EA3 n\u1ED9i dung c\u1EE7a b\u1EA3ng \u0111\xF3.",
  why: () => "Tr\xECnh \u0111\u1ECDc m\xE0n h\xECnh kh\xF4ng th\u1EC3 di\u1EC5n gi\u1EA3i b\u1EA3ng kh\xF4ng c\xF3 c\u1EA5u tr\xFAc ho\xE0n ch\u1EC9nh. Ph\u1EE5 \u0111\u1EC1 b\u1EA3ng l\xE0 th\u1EE9 m\xF4 t\u1EA3 ng\u1EEF c\u1EA3nh v\xE0 th\xF4ng tin n\u1EAFm b\u1EAFt chung v\u1EC1 b\u1EA3ng.",
  link: "https://www.w3.org/TR/WCAG20-TECHS/H39.html",
  linkText: () => "T\xECm hi\u1EC3u th\xEAm v\u1EC1 s\u1EED d\u1EE5ng ch\xFA th\xEDch v\u1EDBi b\u1EA3ng"
};
exports.default = _default;
