// tinymce-a11y-checker/lib/vi/rules/table-header.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _formatMessage = _interopRequireDefault(require("../format-message"));

var _dom = require("../utils/dom");

const _forEach = Array.prototype.forEach;
var _default = {
  id: "table-header",
  test: elem => {
    if (elem.tagName !== "TABLE") {
      return true;
    }

    return elem.querySelector("th");
  },
  data: elem => {
    return {
      header: "none"
    };
  },
  form: () => [{
    label: "C\xE0i \u0111\u1EB7t \u0111\u1EA7u \u0111\u1EC1 b\u1EA3ng",
    dataKey: "header",
    options: [["none", "Kh\xF4ng c\xF3 \u0111\u1EA7u \u0111\u1EC1"], ["row", "H\xE0ng \u0111\u1EA7u \u0111\u1EC1"], ["col", "C\u1ED9t \u0111\u1EA7u \u0111\u1EC1"], ["both", "H\xE0ng v\xE0 c\u1ED9t \u0111\u1EA7u \u0111\u1EC1"]]
  }],
  update: (elem, data) => {
    _forEach.call(elem.querySelectorAll("th"), th => {
      (0, _dom.changeTag)(th, "td");
    });

    if (data.header === "none") {
      return elem;
    }

    const row = data.header === "row" || data.header === "both";
    const col = data.header === "col" || data.header === "both";
    const tableRows = elem.querySelectorAll("tr");

    for (let i = 0; i < tableRows.length; ++i) {
      if (i === 0 && row) {
        _forEach.call(tableRows[i].querySelectorAll("td"), td => {
          const th = (0, _dom.changeTag)(td, "th");
          th.setAttribute("scope", "col");
        });

        continue;
      }

      if (!col) {
        break;
      }

      const td = tableRows[i].querySelector("td");

      if (td) {
        const th = (0, _dom.changeTag)(td, "th");
        th.setAttribute("scope", "row");
      }
    }

    return elem;
  },
  message: () => "B\u1EA3ng ph\u1EA3i c\xF3 \xEDt nh\u1EA5t m\u1ED9t \u0111\u1EA7u \u0111\u1EC1.",
  why: () => "Tr\xECnh \u0111\u1ECDc m\xE0n h\xECnh kh\xF4ng th\u1EC3 di\u1EC5n gi\u1EA3i b\u1EA3ng kh\xF4ng c\xF3 c\u1EA5u tr\xFAc ho\xE0n ch\u1EC9nh. \u0110\u1EA7u \u0111\u1EC1 b\u1EA3ng l\xE0 th\u1EE9 cung c\u1EA5p ph\u01B0\u01A1ng h\u01B0\u1EDBng v\xE0 gi\u1EDBi thi\u1EC7u t\u1ED5ng quan v\u1EC1 n\u1ED9i dung.",
  link: "https://www.w3.org/TR/WCAG20-TECHS/H43.html",
  linkText: () => "T\xECm hi\u1EC3u th\xEAm v\u1EC1 c\xE1c ti\xEAu \u0111\u1EC1 b\u1EA3ng"
};
exports.default = _default;
